internal/pkg/warewulfd/api/node.go

```diff
@@ -123,9 +123,13 @@ func addNode() usecase.Interactor {
 
 	u := usecase.NewInteractor(func(ctx context.Context, input addNodeInput, output *node.Node) error {
 		wwlog.Debug("api.addNode(ID:%v, Node:%+v)", input.ID, input.Node)
+		// registry is the warewulf node "db" yaml file.
 		if registry, err := node.New(); err != nil {
 			return err
 		} else {
+			if _, ok := registry.Nodes[input.ID]; ok {
+				return status.Wrap(fmt.Errorf("node name '%s' already exists", input.ID), status.InvalidArgument)
+			}
```

Comment:
@MatthewHink Please add unit tests for this, too. Since it has already been merged, that will have to be in a new PR.